ci/docker-run.sh

```diff
@@ -76,6 +76,19 @@ ARGS+=(
   --env CRATES_IO_TOKEN
 )
 
+if [[ $BUILDKITE_LABEL == "checks" ]] && echo "$BUILDKITE_BRANCH" | grep -E '^pull/[0-9]+/head$'; then
+  api_base="https://api.github.com/repos/solana-labs/solana/pulls"
+  pr_num=$(echo "$BUILDKITE_BRANCH" | grep -Eo '[0-9]+')
+  branch=$(curl -s "$api_base/$pr_num" | ruby -r json -e 'puts JSON.parse(STDIN.read())["head"]["ref"]')
+  source_repo=$(curl -s "$api_base/$pr_num" | ruby -r json -e 'puts JSON.parse(STDIN.read())["head"]["repo"]["full_name"]')
+
+  if [[ $source_repo == "solana-labs/solana" ]] && echo "$branch" | grep "^dependabot/cargo/"; then
```

Comment:
As this `if` gate is very sensitive to security as this allows the `git push` codepath.